src/main/groovy/nu/studer/gradle/jooq/ToolchainHelper.java

```diff
@@ -0,0 +1,40 @@
+package nu.studer.gradle.jooq;
+
+import nu.studer.gradle.jooq.util.Gradles;
+import org.gradle.api.Task;
+import org.gradle.api.plugins.ExtensionContainer;
+import org.gradle.api.plugins.JavaPluginExtension;
+import org.gradle.api.provider.Property;
+import org.gradle.api.provider.Provider;
+import org.gradle.jvm.toolchain.JavaLauncher;
+import org.gradle.jvm.toolchain.JavaToolchainService;
+import org.gradle.jvm.toolchain.JavaToolchainSpec;
+import org.gradle.process.JavaExecSpec;
+
+/**
+ * This isolates toolchain related types, introduced in 6.7 and above.
+ */
+class ToolchainHelper {
+    private static final String GRADLE_VERSION_WITH_TOOLCHAIN = "7.3.3";
+    private final boolean supportsToolchainAndConfigurationCache;
+    private final Property<Object> launcher;
+
+    public ToolchainHelper(ExtensionContainer extensions, Property<Object> launcher) {
+        this.launcher = launcher;
+        this.supportsToolchainAndConfigurationCache = Gradles.isAtLeastGradleVersion(GRADLE_VERSION_WITH_TOOLCHAIN);
+        if (supportsToolchainAndConfigurationCache) {
+            JavaToolchainSpec toolchain = extensions.getByType(JavaPluginExtension.class).getToolchain();
+            JavaToolchainService service = extensions.getByType(JavaToolchainService.class);
+            Provider<JavaLauncher> defaultLauncher = service.launcherFor(toolchain);
+            launcher.convention(defaultLauncher);
+        }
+    }
+
+    void setExec(JavaExecSpec spec) {
+        if (supportsToolchainAndConfigurationCache && launcher.isPresent() && launcher.get() instanceof JavaLauncher) {
+            spec.setExecutable(((JavaLauncher) launcher.get()).getExecutablePath().getAsFile().getAbsolutePath());
+        } else if (!supportsToolchainAndConfigurationCache && launcher.isPresent()) {
```

Comment:
@alextu IIUC we cannot use toolchain support unless we are on Gradle 7.3.
Is there a way we can allow toolchain usage if the configuration cache is not enable in other versions?

Reply:
As far as I remember it was decided with @etiennestuder to avoid multiple cases. But yes it's technically possible if there's a reliable way (ie cross version) to check if configuration cache is enabled